platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -965,17 +984,18 @@ int _odp_ipsec_sa_replay_precheck(ipsec_sa_t *ipsec_sa, uint32_t seq,
 	return 0;
 }
 
-static inline int ipsec_wslarge_replay_update(ipsec_sa_t *ipsec_sa, uint32_t seq,
+static inline int ipsec_wslarge_replay_update(ipsec_sa_t *ipsec_sa, uint64_t seq,
 					      odp_ipsec_op_status_t *status)
 {
-	uint32_t bucket, wintop_bucket, new_bucket;
-	uint32_t bkt_diff, bkt_cnt, top_seq;
-	uint64_t bit = 0;
+	uint64_t bucket, wintop_bucket, new_bucket;
+	uint64_t bit = 0, top_seq;
+	uint32_t bkt_diff, bkt_cnt;
```

Comment:
The fact the bkt_diff stays uint32_t but get stored the difference between bucket and wintop_bucket looks a bit dubious to me.

At first look it seems like there could be an overflow and then breakage of the subsequent logic. But I suppose maybe such large differences cannot occur here because such packets would be rejected by the authentication check (because we only try with one higher value of the high order part of the ESN). Is that the thinking behind this?

To me this is non-obvious and a bit fragile, so I would suggest just using a 64 bit variable for the bkt_diff too. Or am I missing something?

Reply:
Yes, that's the thought. But will change the bkt_diff as 64-bit variable.